src/promql/src/planner.rs

```diff
@@ -79,10 +82,55 @@ impl<S: ContextProvider> PromPlanner<S> {
 
     pub fn prom_expr_to_plan(&mut self, prom_expr: PromExpr) -> Result<LogicalPlan> {
         let res = match &prom_expr {
-            PromExpr::AggregateExpr { .. } => UnsupportedExprSnafu {
-                name: "Prom Aggregate",
+            PromExpr::AggregateExpr {
```

Comment:
@yuanbohan Maybe the promql parser could define the variants as tuples like datafusion's [LogicalPlan](https://docs.rs/datafusion-expr/latest/datafusion_expr/enum.LogicalPlan.html)
```rust
PromExpr::AggregateExpr(AggregateExpr)
```

It'd be convenient to pass the struct in this variant to other methods
```
PromExpr::AggregateExpr(expr) => self.agg_expr_to_plan(expr),
```

Reply:
good point

Reply:
This will help you https://github.com/GreptimeTeam/promql-parser/pull/23

Reply:
👍